test/unit/atomic_iteration_test.lua

```diff
@@ -134,39 +134,49 @@ end
 -- just check expirationd task continue work after conflicts
 function g.test_mvcc_vinyl_tx_conflict(cg)
     t.skip_if(cg.params.engine ~= 'vinyl', 'Unsupported engine')
-    -- TODO: Remove the line below when tarantool/expirationd#105 is resolved.
-    t.skip('Skipped until tarantool/expirationd#105 is resolved')
+    local tuples_cnt = 10
 
-    for i = 1,10 do
+    for i = 1,tuples_cnt do
         cg.space:insert({i, tostring(i), nil, nil, 0})
     end
 
     local updaters = {}
-    for i = 1,10 do
-        local updater = fiber.create(function()
+    for i = 1,tuples_cnt do
+        local updater = fiber.new(function()
             fiber.name(string.format("updater of %d", i), { truncate = true })
-            while true do
+            box.atomic(function()
                 cg.space:update({i}, { {"+", 5, 1} })
```

Comment:
You don't need to wrap update to box.atomic. Single update operation is in transaction anyway.

Reply:
It also looks strange because you use atomic_iteration = true... Seems you wrap transaction into transaction (it's strange that it doesn't lead to any errors).

Reply:
> You don't need to wrap update to box.atomic. Single update operation is in transaction anyway.

Thanks for the simplification. I originally had a different idea, a bit confused as a result.

Reply:
> It also looks strange because you use atomic_iteration = true... Seems you wrap transaction into transaction (it's strange that it doesn't lead to any errors).

```
-- @boolean[opt] options.atomic_iteration

False (default) to process each tuple as a single transaction and true
to process tuples from each batch in a single transaction.
```

So we have a `select tuples + delete` tx (from expirationd). It is in conflict with update transactions from fibers. Everything looks good for me.

Reply:
Yes, you are right. It's different transactions. I was a little confused.